renovate.json

```diff
@@ -0,0 +1,26 @@
+{
+    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+    "rebaseWhen": "behind-base-branch",
+    "dependencyDashboard": true,
+    "pip-compile": {
+        "fileMatch": [
+            "(^|/)requirements.*\\.in$"
+        ],
+        "lockFileMaintenance": {
+            "schedule": null
+        }
+    },
+    "packageRules": [
```

Comment:
You might want to adjust the merge strategy to use branches - otherwise you'll get a PR for every dep change. It's also possible to group dependencies and schedule weekly, for example.

Not completely convinced I've got it sorted yet, but here's my attempt at keeping noise down: https://github.com/jnsgruk/parca-k8s-operator/blob/main/.github/renovate.json

Reply:
Ah so if `automergeType!=branch`, it'll try to handle each one of the dependencies in the group separately?  I was thinking the entire group would get lumped together in a single PR that can then be auto-merged

Reply:
Grouping is separate - the branch merge type means it'll just push changes to a branch (without PR), and if alll the checks pass on the branch it'll merge to main (without a PR)

I'm not 💯 sure I nailed the config so very happy for suggestions, but last night two of my charms got updated and released to edge with no notification spam aapart from one notification for each GitHub Release :)

Reply:
Love it thanks

Reply:
Lol I would swear I pushed it MB